contrib/demo/workspaceKubectlPlugin-script/workspaceKubectlPlugin

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DEMO_DIR="$( dirname "${BASH_SOURCE[0]}" )"
+DEMOS_DIR="$(cd "${DEMO_DIR}/.."; pwd)"
+
+. ${DEMOS_DIR}/demo-magic
+
+ROOT_DIR="$( cd ${DEMOS_DIR}/../.. && pwd)"
+
+TYPE_SPEED=30
+#PROMPT_AFTER=1
+DEMO_PROMPT="☸️ $ "
+
+function pause() {
+  if [[ -n "${NO_WAIT}" ]]; then
+    sleep 2
+  else
+    if [[ -n "${1-}" ]]; then
+      sleep "$1"
+    else
+      wait
+    fi
+  fi
+}
+
+export KUBECONFIG=${KUBECONFIG:-${ROOT_DIR}/.kcp/admin.kubeconfig}
+if ! kubectl get namespaces &>/dev/null; then
+  echo "kcp server not started, run 'bin/kcp start'"
+  exit 1
+fi
+
+kubectl create secret generic kubeconfig --from-file=kubeconfig=${KUBECONFIG}
+kubectl apply -f ${DEMO_DIR}/workspace-shard.yaml
+
+touch myKubeConfig
+export PATH=$PATH:${ROOT_DIR}/bin
+export KUBECONFIG=$(pwd)/myKubeConfig
+
+kubectl kcp workspace --token user-1-token --workspace-directory-insecure-skip-tls-verify list
+
+clear
+
+pe "kubectl kcp workspace --help"
+
+pause
+clear
+
+pe "echo \${KUBECONFIG}"
+pe "more ${KUBECONFIG}"
+pe ""
+
+pause
+clear
+
+# creates the cluster roles + bindings at the workspace level + workspace content level.
+
+pe "kubectl kcp workspace --token user-1-token create workspace11"
```

Comment:
This demo was really hard for me to follow. Can we do some or all of the following:

 - prefix workspace names with their users, so `workspace11` becomes `user-1-workspace-1`
 - not mix current contexts that are user-specific with other tokens

Especially the second point - it's very opaque what is happening here. I believe with OIDC you *only* authenticate with the token, right? But the workspace context is user-specific? It's not clear.